src/release-specification.ts

```diff
@@ -301,6 +300,72 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release packages with breaking changes alongside their dependents
+      if (
+        versionSpecifierOrDirective === 'major' ||
+        (isValidSemver(versionSpecifierOrDirective) &&
+          diff(
+            pkg.validatedManifest.version,
+            versionSpecifierOrDirective as string,
+          ) === 'major')
+      ) {
+        const missingDependents = Object.values(
+          project.workspacePackages,
+        ).filter((dependent) => {
+          const { dependencies, peerDependencies } =
+            dependent.unvalidatedManifest;
+          const isDependent =
+            (dependencies && hasProperty(dependencies, packageName)) ||
+            (peerDependencies && hasProperty(peerDependencies, packageName));
+
+          if (!isDependent) {
+            return false;
+          }
+
+          const dependentVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[
+              dependent.validatedManifest.name
+            ];
+
+          return (
+            dependentVersionSpecifierOrDirective !== SKIP_PACKAGE_DIRECTIVE &&
+            dependentVersionSpecifierOrDirective !==
+              INTENTIONALLY_SKIP_PACKAGE_DIRECTIVE &&
+            !hasProperty(
+              IncrementableVersionParts,
+              dependentVersionSpecifierOrDirective,
+            ) &&
+            !isValidSemver(dependentVersionSpecifierOrDirective)
+          );
+        });
+
+        if (missingDependents.length > 0) {
+          errors.push({
+            message: [
+              `The following packages, which depends on released package ${packageName}, are missing.`,
+              missingDependents
+                .map((dependent) => `  - ${dependent.validatedManifest.name}`)
+                .join('\n'),
+              " Consider including them in the release spec so that they won't break in production.",
```

Comment:
Nit: Maybe better to describe this as a compatibility problem. It was a bit unclear from reading this _how_ they might break things, but "compatibility" is a way to get the point across without going into detail.
```suggestion
              ` Consider including them in the release spec so that they are compatible with the new ${packageName} version.`,
```